openstudiocore/src/model/SiteGroundTemperatureDeep.cpp

```diff
@@ -0,0 +1,653 @@
+/**********************************************************************
+ *  Copyright (c) 2008-2016, Alliance for Sustainable Energy.
+ *  All rights reserved.
+ *
+ *  This library is free software; you can redistribute it and/or
+ *  modify it under the terms of the GNU Lesser General Public
+ *  License as published by the Free Software Foundation; either
+ *  version 2.1 of the License, or (at your option) any later version.
+ *
+ *  This library is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ *  Lesser General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Lesser General Public
+ *  License along with this library; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ **********************************************************************/
+
+#include <vector>
+#include "SiteGroundTemperatureDeep.hpp"
+#include "SiteGroundTemperatureDeep_Impl.hpp"
+
+#include "Model.hpp"
+#include "Model_Impl.hpp"
+#include "Site.hpp"
+#include "Site_Impl.hpp"
+
+#include <utilities/idd/OS_Site_GroundTemperature_Deep_FieldEnums.hxx>
+#include <utilities/idd/IddEnums.hxx>
+
+#include "../utilities/core/Assert.hpp"
+
+namespace openstudio {
+namespace model {
+
+namespace detail {
+
+  SiteGroundTemperatureDeep_Impl::SiteGroundTemperatureDeep_Impl(
+      const IdfObject& idfObject,
+      Model_Impl* model,
+      bool keepHandle)
+    : ModelObject_Impl(idfObject,model,keepHandle)
+  {
+    OS_ASSERT(idfObject.iddObject().type() == SiteGroundTemperatureDeep::iddObjectType());
+  }
+
+  SiteGroundTemperatureDeep_Impl::SiteGroundTemperatureDeep_Impl(const openstudio::detail::WorkspaceObject_Impl& other,
+                                                                                       Model_Impl* model,
+                                                                                       bool keepHandle)
+    : ModelObject_Impl(other,model,keepHandle)
+  {
+    OS_ASSERT(other.iddObject().type() == SiteGroundTemperatureDeep::iddObjectType());
+  }
+
+  SiteGroundTemperatureDeep_Impl::SiteGroundTemperatureDeep_Impl(const SiteGroundTemperatureDeep_Impl& other,
+                                                                                       Model_Impl* model,
+                                                                                       bool keepHandle)
+    : ModelObject_Impl(other,model,keepHandle)
+  {}
+
+  boost::optional<ParentObject> SiteGroundTemperatureDeep_Impl::parent() const
+  {
+    boost::optional<Site> result = this->model().getOptionalUniqueModelObject<Site>();
```

Comment:
If Site is the parent then you should add this to Site's children:

https://github.com/NREL/OpenStudio/blob/develop/openstudiocore/src/model/Site.cpp#L97